source/pkger/pkger.go

```diff
@@ -0,0 +1,67 @@
+package pkger
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/golang-migrate/migrate/v4/source"
+	"github.com/golang-migrate/migrate/v4/source/httpfs"
+	"github.com/markbates/pkger/pkging"
+)
+
+func init() {
+	source.Register("pkger", &driver{})
+}
+
+// Instance is an implementation of http.FileSystem backed by an instance of
+// pkging.Pkger.
+type Instance struct {
+	pkging.Pkger
+
+	// Path is the relative path location of the migrations. It is passed to
+	// httpfs.PartialDriver.Init. If unset "/" is used as all paths are
+	// absolute.
+	Path string
+}
+
+// Open implements http.FileSystem.
+func (p *Instance) Open(name string) (http.File, error) {
+	f, err := p.Pkger.Open(name)
+	if err != nil {
+		return nil, err
+	}
+	return f.(http.File), nil
+}
+
+type driver struct {
+	httpfs.PartialDriver
+}
+
+// Open implements source.Driver. NOT IMPLEMENTED.
+func (d *driver) Open(url string) (source.Driver, error) {
+	return nil, fmt.Errorf("not yet implemented")
+}
+
+// WithInstance returns a source.Driver that is backed by an Instance.
+func WithInstance(instance interface{}) (source.Driver, error) {
+	if _, ok := instance.(*Instance); !ok {
+		return nil, fmt.Errorf("expects *Instance")
+	}
+
+	p := instance.(*Instance)
+
+	if p.Path == "" {
+		p.Path = "/"
+	}
+
+	var fs http.FileSystem
+	var ds driver
+
+	fs = p
+
+	if err := ds.Init(fs, p.Path); err != nil {
```

Comment:
`Init()` should be called with an instance of `pkging.Pkger`

Reply:
Does this still apply after reading the explanation of why there are two structs?

`pkging.Pkger` does not implement `http.FileSystem`. Do you mean to call `Init` with `*Pkger` (previously `*Instance`)? It is in fact already doing that. If it's unclear, perhaps I could remove
```
var fs http.FileSystem
fs = instance
```
and pass `instance` directly to `Init`. Would that make it clearer?